LinuxCommunicator/HostsFile/HostsFileManager.cs

```diff
@@ -0,0 +1,215 @@
+        /// </summary>
+        private static readonly Regex Comment = new Regex(@"^#(?<1>.*)$");
+
+        /// <summary>
+        /// Regular expression for a parameter value in the comments section.
+        /// </summary>
+        private static readonly Regex CommentParameter = new Regex(@"^#\s*(?<parameter>[\w\p{Lm}\p{Nl}\p{Cf}\p{Mn}\p{Mc}\.]+)\s*=\s*(?<value>[\w\p{Lm}\p{Nl}\p{Cf}\p{Mn}\p{Mc}\.]+)");
+
+        /// <summary>
+        /// Regular expression for a ip entry in the text file.
+        /// </summary>
+        private static readonly Regex IpEntry = new Regex(@"^(?<ip>[0-9\.]+)\s+(?<dnsName>[^\s#]+)(\s+#(?<comment>.*))?");
+
+        /// <summary>
+        /// The ip address entries in the host file.
+        /// </summary>
+        private List<HostEntry> managedEntries = new List<HostEntry>();
+
+        /// <summary>
+        /// The path to the hosts file
+        /// </summary>
+        private string filepath;
+
+        /// <summary>
+        /// The timer to reload hosts file
+        /// </summary>
+        private Timer reloadTimer;
+
+        /// <summary>
+        /// The last modified date of the current loaded hosts file
+        /// </summary>
+        private DateTime lastModified = DateTime.MinValue;
+
+        /// <summary>
+        /// The unique update ID
+        /// </summary>
+        public Guid UpdateId
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// This constructor initializes the parser on a specific hosts file
+        /// </summary>
+        /// <param name="filepath"></param>
+        public HostsFileManager(string filepath)
+        {
+            this.filepath = filepath;
+            this.reloadTimer = new Timer(new TimerCallback(ReloadTimerEvent), null, 0, Timeout.Infinite);
+        }
+
+        /// <summary>
+        /// Gets the list of HPC managed entries in the host file.
+        /// </summary>
+        public List<HostEntry> ManagedEntries
+        {
+            get { return this.managedEntries; }
+        }
+
+        /// <summary>
+        /// Gets or sets a flag indicating whether the file is HPC managed
+        /// </summary>
+        public Boolean ManageFile
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Loads the contents of an existing host file. Will clear
+        /// the current configuration of this instance.
+        /// </summary>
+        /// <param name="path"></param>
+        public void ReloadTimerEvent(object param)
+        {
+
+            try
+            {
+                FileInfo fileInfo = new FileInfo(this.filepath);
+                if(!fileInfo.Exists)
+                {
+                    LinuxCommunicator.Instance.Tracer.TraceInfo("[HostsFileManager] The hosts file doesn't exists: {0}", this.filepath);
+                    return;
+                }
+
+                if(fileInfo.LastWriteTimeUtc <= this.lastModified)
```

Comment:
How about write !fileInfo.Exists || fileInfo.LastWriteTimeUtc <= this.lastModified and combine the logs as "Skill the reload because Exists {0}, LastWriteTime {1}, lastReadTime {2}